src/Components/SysTable/constants.js

```diff
@@ -100,22 +106,41 @@ export const columns = [
     }
 ];
 
-export const mergedColumns = (defaultColumns) =>
-    columns.map((column) => {
+export const mergedColumns = (defaultColumns) => {
+    return columns.map((column) => {
         const isStringCol = typeof column === 'string';
-        const key = isStringCol ? column : column.key;
+
         const defaultColumn = defaultColumns.find(
-            (defaultCol) => defaultCol.key === key
+            (defaultCol) => defaultCol.key === column.key
         );
-        return {
-            ...defaultColumn,
-            ...(isStringCol ? { key: column } : column),
-            props: {
-                ...defaultColumn?.props,
-                ...column?.props
-            }
-        };
-    });
+        //this step is necessary so we could compare the default columns from column object at the top of the file
+        //with the columns that come from the inventory, so if inventory hide a column behind feature flag
+        //it will not appear when the flag is turned off
+        return column.requiresDefault && defaultColumn === undefined ?
+            undefined : {
+                ...(isStringCol ? { key: column } : column),
+                ...defaultColumn,
+                ...(column.key === 'groups'
+                    ? {
+                        renderFunc: (value) =>
+                            isEmpty(value?.groups) ? (
+                                'N/A'
+                            ) : (
+                                <a href={`/insights/inventory/groups/${value?.groups[0].id}`}>
+                                    {
+                                        value?.groups[0].name // currently, one group at maximum is supported
+                                    }
+                                </a>
```

Comment:
Please, add the TODO note about rewriting the link component with react-router v6 upgrade